runtimes/node-18.0/src/server.js

```diff
@@ -1,23 +1,26 @@
-const fs = require("fs");
 const micro = require("micro");
-const util = require("util");
 const { text: parseText, json: parseJson, send } = require("micro");
+const Logger = require("./logger");
 
 const USER_CODE_PATH = '/usr/local/server/src/function';
 
 const server = micro(async (req, res) => {
+    const logger = new Logger(req.headers[`x-open-runtimes-logging`], req.headers[`x-open-runtimes-log-id`]);
+
     try {
-        await action(req, res);
+        await action(logger, req, res);
     } catch(e) {
-        const logs = [];
-        const errors = [e.stack || e];
-        res.setHeader('x-open-runtimes-logs', encodeURIComponent(logs.join('\n')));
-        res.setHeader('x-open-runtimes-errors', encodeURIComponent(errors.join('\n')));
+        logger.write(e, Logger.TYPE_ERROR);
+        res.setHeader('x-open-runtimes-log-id', logger.id);
+        await logger.end();
+
         return send(res, 500, '');
+    } finally {
+        await logger.end();
```

Comment:
since finally executes in the end, you can also remove the logger.end() in Line 205